src/model/mod.rs

```diff
@@ -0,0 +1,385 @@
+        }
+    }
+    /// Get the advance rate of the Model
+    ///
+    /// # Returns
+    /// advance (usize): The advance rate of the Model
+    pub fn get_advance(&self) -> usize {
+        self.advance
+    }
+
+    /// Get the march rate of the Model
+    ///
+    /// # Returns
+    /// march (usize): The march rate of the Model
+    pub fn get_march(&self) -> usize {
+        self.march
+    }
+
+    /// Get the discipline of the Model
+    ///
+    /// # Returns
+    /// discipline (usize): The discipline of the Model
+    pub fn get_discipline(&self) -> usize {
+        self.discipline
+    }
+
+    /// Get the health point of the Model
+    ///
+    /// # Returns
+    /// health_point (usize): The health point of the Model
+    pub fn get_health_point(&self) -> usize {
+        self.health_point
+    }
+
+    /// Get the defense of the Model
+    ///
+    /// # Returns
+    /// defense (usize): The defense of the Model
+    pub fn get_defense(&self) -> usize {
+        self.defense
+    }
+
+    /// Get the resilience of the Model
+    ///
+    /// # Returns
+    /// resilience (usize): The resilience of the Model
+    pub fn get_resilience(&self) -> usize {
+        self.resilience
+    }
+
+    /// Get the armour of the Model
+    ///
+    /// # Returns
+    /// armour (usize): The armour of the Model
+    pub fn get_armour(&self) -> usize {
+        self.armour
+    }
+
+    /// Get the aegis of the Model
+    ///
+    /// # Returns
+    /// aegis (usize): The aegis of the Model
+    pub fn get_aegis(&self) -> usize {
+        self.aegis
+    }
+
+    /// Get the attack of the Model
+    ///
+    /// # Returns
+    /// attack (usize): The attack of the Model
+    pub fn get_attack(&self) -> usize {
+        self.attack
+    }
+
+    /// Get the offensive of the Model
+    ///
+    /// # Returns
+    /// offensive (usize): The offensive of the Model
+    pub fn get_offensive(&self) -> usize {
+        self.offensive
+    }
+
+    /// Get the strength of the Model
+    ///
+    /// # Returns
+    /// strength (usize): The strength of the Model
+    pub fn get_strength(&self) -> usize {
+        self.strength
+    }
+
+    /// Get the armour penetration of the Model
+    ///
+    /// # Returns
+    /// armour_penetration (usize): The armour penetration of the Model
+    pub fn get_armour_penetration(&self) -> usize {
+        self.armour_penetration
+    }
+
+    /// Get the agility of the Model
+    ///
+    /// # Returns
+    /// agility (usize): The agility of the Model
+    pub fn get_agility(&self) -> usize {
+        self.agility
+    }
+}
+
+/// Struct containing all the information about a single Modifier for a Model
+///
+/// # Attributes
+/// stat (Stats): Stat modified by the Modifier
+///
+/// nb_dice (usize): The number of dice
+///
+/// requirements (Vec<String>): The requirements for the Modifier to apply (as an array of flags)
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Modifier {
+    stat: Stats,
```

Comment:
It's very inefficient to create another struct maybe change the type of the variables in Stats (this will lead to huge modifications in the current code because we used u8 instead of i8 because our values couldn't be negatives) so maybe add a new element in the modifier to specify if it's a buff or a debuff

Reply:
What about the case where you're being buff for a stat and debug for another ?
Like, -> Hey you have a boost of 2 attack but it leads to a loss of 1 def
 just a boolean not work on this case

Reply:
The model struct have a Vec of Modifiers so vec[0] -> attack 2 bool = true (for bonus) and vec[1] -> def 1 bool = false (for malus)

Reply:
The modifiers vec should be enough. Are we checking that the modifiers are modifying the stats when the bool is true ? If yes then we should be good, if not we should properly deal with it in the next sprint

Reply:
For now we don't use modifiers at all

Reply:
ok then, go for a boolean value

Reply:
Done